specification/cloudhealth/CloudHealth.Management/main.tsp

```diff
@@ -0,0 +1,843 @@
+
+  @doc("Unhealthy threshold for aggregating the propagated health state of child dependencies. Can be either an absolute number that is greater than 0, or a percentage between 1-100%. The entity will be considered unhealthy when the number of not healthy child dependents (unhealthy, degraded, unknown) is equal to or above the threshold value. Must only be set when AggregationType is 'Thresholds'.")
+  unhealthyThreshold?: string;
+}
+
+@doc("Aggregation type for child dependencies.")
+union DependenciesAggregationType {
+  string,
+
+  @doc("Default behavior: Worst child health state is propagated.")
+  WorstOf: "WorstOf",
+
+  @doc("Based on configurable thresholds.")
+  Thresholds: "Thresholds",
+}
+
+@doc("A grouping of signal assignments and link to a data source")
+model SignalAssignmentGroup {
+  @doc("Signal definitions which are assigned to this signal group. All assignments are combined with an OR operator.")
+  @maxItems(50)
+  @OpenAPI.extension("x-ms-identifiers", #["signalDefinitions"])
+  signalAssignments?: SignalAssignment[];
+
+  @doc("Reference to the name of the authentication setting which is used for querying the data source")
+  authenticationSetting: stringResourceName;
+}
+
+@doc("Group of signal definition assignments")
+model SignalAssignment {
+  @doc("Signal definitions referenced by their names. All definitions are combined with an AND operator.")
+  // @maxItems(50) - until the backend supports correlation, we need to limit this to 1
+  @maxItems(1)
+  @minItems(1)
+  signalDefinitions: stringResourceName[];
+}
+
+@doc("A grouping of signal assignments for an Azure resource")
+model AzureResourceSignalGroup is SignalAssignmentGroup {
+  @doc("Azure resource ID")
+  azureResourceId: armResourceIdentifier<[]>;
+}
+
+@doc("A grouping of signal assignments for a Log Analytics Workspace")
+model LogAnalyticsSignalGroup is SignalAssignmentGroup {
+  @doc("Log Analytics Workspace resource ID")
+  logAnalyticsWorkspaceResourceId: LogAnalyticsWorkspaceResourceId;
+}
+
+@doc("A grouping of signal assignments for a Azure Monitor Workspace")
+model AzureMonitorWorkspaceSignalGroup is SignalAssignmentGroup {
+  @doc("Azure Monitor workspace resource ID")
+  azureMonitorWorkspaceResourceId: AzureMonitorWorkspaceResourceId;
+}
+
+@doc("Type of impact an entity has on health state propagation")
+union EntityImpact {
+  string,
+
+  @doc("Standard impact")
+  Standard: "Standard",
+
+  @doc("Limited impact")
+  Limited: "Limited",
+
+  @doc("Suppressed impact")
+  Suppressed: "Suppressed",
+}
+
+@doc("Health state of an entity")
+union HealthState {
+  string,
+
+  @doc("Healthy status")
+  Healthy: "Healthy",
+
+  @doc("Degraded status")
+  Degraded: "Degraded",
+
+  @doc("Error status (Unhealthy)")
+  Error: "Error",
+
+  @doc("Unknown status")
+  Unknown: "Unknown",
+
+  @doc("Deleted status")
+  Deleted: "Deleted",
+}
+
+@doc("Visual position of the entity")
+model EntityCoordinates {
+  @doc("X Coordinate")
```

Comment:
>Coordinate [](http://example.com/codeflow?start=10&length=10)

are there units?

Reply:
no. Just X and Y on a canvas. The canvas is not explicitly set to pixels, inch etc.